bind-mount.c

```diff
@@ -455,10 +455,71 @@ bind_mount (int           proc_fd,
               /* If we can't read the mountpoint we can't remount it, but that should
                  be safe to ignore because its not something the user can access. */
               if (errno != EACCES)
-                return 5;
+                return BIND_MOUNT_ERROR_REMOUNT_SUBMOUNT;
             }
         }
     }
 
-  return 0;
+  return BIND_MOUNT_SUCCESS;
+}
+
+void
+die_with_bind_result (bind_mount_result res,
+                      int               saved_errno,
+                      const char       *format,
+                      ...)
+{
+  va_list args;
+  bool want_errno = TRUE;
+
+  fprintf (stderr, "bwrap: ");
+
+  va_start (args, format);
+  vfprintf (stderr, format, args);
+  va_end (args);
+
+  fprintf (stderr, ": ");
+
+  switch (res)
+    {
+      case BIND_MOUNT_ERROR_MOUNT:
+        fprintf (stderr, "Unable to mount source on destination");
+        break;
+
+      case BIND_MOUNT_ERROR_REALPATH_DEST:
+        fprintf (stderr, "realpath(destination)");
+        break;
+
+      case BIND_MOUNT_ERROR_REOPEN_DEST:
+        fprintf (stderr, "open(destination, O_PATH)");
+        break;
+
+      case BIND_MOUNT_ERROR_READLINK_DEST_PROC_FD:
+        fprintf (stderr, "readlink(/proc/self/fd/<destination>)");
+        break;
+
+      case BIND_MOUNT_ERROR_FIND_DEST_MOUNT:
+        fprintf (stderr, "Unable to find destination in mount table");
+        want_errno = FALSE;
```

Comment:
Though I guess this is a wrinkle so it'd need to return two values.  (Or...maybe just reset `errno = 0` and don't print it if zero?)

Reply:
I'm already not a fan of the "action at a distance" API that is `errno`, so having helper functions edit `errno` as a side-effect seems nasty to me.

I was half-tempted to reinvent the `GError`/`DBusError` pattern here, [like I did in libcapsule](https://gitlab.collabora.com/vivek/libcapsule/-/blob/master/utils/utils.h#L145), but held off on doing that *right* now.

Reply:
I went for "return two values".